src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -67,190 +68,186 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
-            }
-        }
-
-        private static string GetReceiverTypeName(ITypeSymbol typeSymbol)
-        {
-            switch (typeSymbol)
-            {
-                case INamedTypeSymbol namedType:
-                    return namedType.MetadataName;
-
-                case IArrayTypeSymbol arrayType:
-                    var elementType = arrayType.ElementType;
-                    while (elementType is IArrayTypeSymbol symbol)
+                    else
                     {
-                        elementType = symbol.ElementType;
+                        bestSymbol = symbol;
+                        overloadCount = 1;
                     }
 
-                    var elementTypeName = GetReceiverTypeName(elementType);
-
-                    // We do not differentiate array of different kinds sicne they are all represented in the indices as "NonArrayElementTypeName[]"
-                    // e.g. int[], int[][], int[,], etc. are all represented as "int[]", whereas array of complex type such as T[] is "[]".
-                    return elementTypeName + FindSymbols.Extensions.ArrayReceiverTypeNameSuffix;
-
-                default:
-                    // Complex types are represented by "";
-                    return FindSymbols.Extensions.ComplexReceiverTypeName;
+                    overloadMap[(containingNamespacename, symbol.Name, symbol.Arity > 0)] = (bestSymbol, overloadCount);
+                }
             }
-        }
 
-        public static async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)> GetUnimportedExtensionMethodsInCurrentProcessAsync(
-            Document document,
-            int position,
-            ITypeSymbol receiverTypeSymbol,
-            ISet<string> namespaceInScope,
-            bool forceIndexCreation,
-            CancellationToken cancellationToken)
-        {
-            var counter = new StatisticCounter();
-            var ticks = Environment.TickCount;
+            // Then convert symbols into completion items
+            using var _3 = ArrayBuilder<SerializableImportCompletionItem>.GetInstance(out var itemsBuilder);
 
-            var indicesResult = await TryGetIndicesAsync(
-                document.Project, forceIndexCreation, cancellationToken).ConfigureAwait(false);
+            foreach (var (methodData, overloadSymbols) in overloadMap)
+            {
+                // To display the count of of additional overloads, we need to substract total by 1.
+                var item = CreateItem(overloadSymbols.bestSymbol, methodData.containingNamespace, additionalOverloadCount: overloadSymbols.overloadCount - 1, cancellationToken);
+                itemsBuilder.Add(item);
+            }
 
-            // Don't show unimported extension methods if the index isn't ready.
-            if (!indicesResult.HasResult)
+            // If we don't have all the indices available already, queue a backgrounds task to create them.
+            if (!indicesComplete)
```

Comment:
what sets this to false... it looks like it's just true above, and not change between the declaration and this line.

Reply:
n/m.